### ISSUE TITLE
Dynamic args don't work in DO/COMMAND

### ISSUE THREAD

**Issue by jclangst** [earthly/earthly#1159]
Currently, you can provide dynamic arguments to build: https://docs.earthly.dev/docs/earthfile#build

It would be amazing to be able to pass dynamic arguments into the DO/COMMAND structure.

Example:

```Dockerfile
DO +COMMAND --ARG=$(echo "I am dynamic")
```

It appears to already do the evaluation, but it provides an error like the following:

`Error: build target: build main: bkClient.Build: failed to solve: Earthfile line 56:2 parse flag args: invalid argument I am dynamic)`

So perhaps this is already supposed to be supported and is a bug?

**Comment by jclangst** [comment]
For reference to others potentially looking for this, I was able to emulate my goal behavior by splitting into multiple steps:

```dockerfile
ARG DYNAMIC_ARG=$(echo "I am dynamic")
DO +COMMAND --ARG=$DYNAMIC_ARG
```

**Comment by jclangst** [comment]
For reference, this is on `0.5.22`.

**Comment by vladaionescu** [comment]
Seems like a bug indeed.

**Comment by alexcb** [comment]
This has been fixed; it works in v0.6.9 (and v0.7.1 too).
